elasticsearch/src/main/java/com/netflix/conductor/elasticsearch/EmbeddedElasticSearch.java

```diff
@@ -0,0 +1,47 @@
+package com.netflix.conductor.elasticsearch;
+
+import com.netflix.conductor.service.Lifecycle;
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public interface EmbeddedElasticSearch extends Lifecycle {
+    Logger logger = LoggerFactory.getLogger(EmbeddedElasticSearch.class);
+    String ES_PATH_DATA = "path.data";
+    String ES_PATH_HOME = "path.home";
+
+    int DEFAULT_PORT = 9200;
+    String DEFAULT_CLUSTER_NAME = "elasticsearch_test";
+    String DEFAULT_HOST = "127.0.0.1";
+    String DEFAULT_SETTING_FILE = "embedded-es.yml";
+
+    default void cleanDataDir(String path) {
+        try {
+            logger.info("Deleting contents of data dir {}", path);
+            File f = new File(path);
+            if (f.exists()) {
+                FileUtils.cleanDirectory(new File(path));
+            }
```

Comment:
I think this is a bug in the original code, but we want `FileUtils.cleanDirectory(f);` here, instead of opening the file again. The current implementation, besides redundant, could be racy!

Reply:
I don't there is a potential race, but it is still worth cleaning up.  File objects are immutable and while some of the operations have side affects, they don't change anything about the path/object itself.